### ISSUE TITLE
spack spec --yaml has been broken on develop since April

### ISSUE THREAD

**Issue by chissg** [spack/spack#8790]
`spack spec --yaml <spec>` used to produce a machine-readable representation of the spec information for a package and its dependencies. Now, it just produces an error:
```
spack spec --yaml boost
==> Error: argument of type 'module' is not iterable
```

Use of `git bisect`identifies commit 4a54178ff as the culprit.

### Information on your system

Scientific Linux 7.4 with Python 2.7.5 using Spack at the head of develop.

FAO: @adamjstewart @tgamblin

**Comment by chissg** [comment]
Also FYI: @lgarren @amundson

**Comment by chissg** [comment]
I would note respectfully that this is a showstopper for SpackDev, so if there's a quick fix for this we'd be grateful.

**Comment by alalazo** [comment]
@tgamblin Out of curiosity: was really 4a54178 the culprit? I am quite surprised that modifying a package we can break a command for unrelated packages.

**Comment by tgamblin** [comment]
No, it was c615d2be. @chissg posted the right commit on Slack -- not sure why he put that one here.

**Comment by chissg** [comment]
@tgamblin Bad cut and paste thinko, I think: 4a54178 was last-known-good, i.e. the starting point for the bisect. Sorry for the confusion. Should I edit the original post to correct the record?

**Comment by tgamblin** [comment]
I think it’s no big deal :). It’s fixed!
